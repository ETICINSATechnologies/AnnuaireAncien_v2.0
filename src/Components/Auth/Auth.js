class Auth {
    static connect(jwt){
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(){
        sessionStorage.removeItem('token')
    }

    static isConnected(){
        return !!sessionStorage.getItem('token');
    }

    static getToken() {
        return sessionStorage.getItem('token')
    }

    static getPayload(){
        return sessionStorage.getItem('token');
    }

    static addCorrectButton(buttons){
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
