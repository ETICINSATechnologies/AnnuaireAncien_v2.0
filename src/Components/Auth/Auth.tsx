class Auth {
    static connect(jwt: string) {
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(): void {
        sessionStorage.removeItem('token')
    }

    static isConnected(): boolean {
        return !!sessionStorage.getItem('token');
    }

    static getToken(): string {
        return sessionStorage.getItem('token') as string;
    }

    static addCorrectButton(buttons: Array<string>) {
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
