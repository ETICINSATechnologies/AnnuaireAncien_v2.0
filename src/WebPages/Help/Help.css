.Help {
    display: grid;
    width: 100%;
    height: 100%;
    text-align: center;
}

.Help .help_section {
    display: flex;
    flex-direction: column;
    grid-template-rows: 1fr 1fr 1fr 1fr;
    margin: 3vmin auto;
    width: 95%;
}

.Help .help_section * {
    text-align: center;
    justify-self: center;
    margin: 0 auto;
    font-size: 3rem;
}

.Help .byetic {
    color: #000000;
    font-size: 2rem;
    text-align: center;
}

.Help .logoETIC {
    width: 220px;
    margin-bottom: 0px;
}

.Help .actu {
    font-size: 1.5rem;
    margin-top: 0px;
}

.Help .social * {
    text-decoration: none;
    text-align: center;
    width: 50px;
    margin-right: 10px;
    margin-left: 10px;
    margin-top: 15px;
}

.Help .explications {
    color: #0c7466;
    font-size: 1rem;
    text-align: center;
}

.Help .contact.toCopy {
    cursor: pointer;
    color: #009e11;
    background: white;
    border: 0;
    text-decoration: underline;
    font-size: 1rem;
}

.Help .contact.copied {
    display: inline-block;
    animation-duration: 2s;
    animation-name: display;
    color: white;
    font-size: 2vw;
    padding: 0.3vmin;
    margin: 1vw auto;
    width: auto;
}

@keyframes display {
    0% {
        color: white;
        background-color: white;
    }
    30% {
        color: black;
        background-color: orange;
    }
    100% {
        color: white;
        background-color: white;
    }
}