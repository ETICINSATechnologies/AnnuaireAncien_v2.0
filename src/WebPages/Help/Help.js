import React, {Component} from 'react';
import './Help.css';
import Header from "../../Components/Header/Header";
import Nav from "../../Components/Nav/Nav";
import Auth from "../../Components/Auth/Auth";

class Help extends Component {
    render() {
        let activeButton = ['home'];
        if (Auth.isConnected()) {
            activeButton.push('search');
            activeButton.push('profile');
        }
        activeButton = Auth.addCorrectButton(activeButton);

        return (
            <React.Fragment>
                <Header/>
                <Nav buttons={activeButton}> </Nav>
                <div className="Help">
                    <section className="help_section">
                        <div className="div1">
                            Cette application a été développée par
                        </div>
                        <a className="div2" href="http://etic-insa.com/"> ETIC INSA Technologies </a>
                        <div className="div3">
                            Elle permet de rechercher les anciens membres de la Junior en fonction de divers critères
                            comme leur
                            nom,
                            prénom, entreprise pour laquelle ils travaillent ou encore leur année de mandat
                            Si vous avez des remarques ou suggestions, merci d'en faire par à
                            <button className="div4"> contact@etic-insa.com </button>
                            <div className="div5"> adresse copié !</div>
                        </div>
                    </section>
                </div>
            </React.Fragment>
        );
    }
}

export default Help;
