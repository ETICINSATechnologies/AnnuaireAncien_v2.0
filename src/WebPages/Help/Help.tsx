import React from 'react';
import './Help.css';
import Header from "../../Components/Header/Header";
import Nav from "../../Components/Nav/Nav";
import Auth from "../../Components/Auth/Auth";
import CopyContent from "../../Components/CopyContent/CopyContent";


const Help: React.SFC<{}> = () => {
    let activeButton = ['home'];
    if (Auth.isConnected()) {
        activeButton.push('search');
        activeButton.push('profile');
    }
    activeButton = Auth.addCorrectButton(activeButton);

    return (
        <React.Fragment>
            <Header/>
            <Nav buttons={activeButton}> </Nav>
            <div className="Help">
                <section className="help_section">
                    <div className="div1">
                        Cette application a été développée par
                    </div>
                    <a className="div2" href="http://etic-insa.com/"> ETIC INSA Technologies </a>
                    <div className="div3">
                        Elle permet de rechercher les anciens membres de la Junior en fonction de divers critères
                        comme leur nom, prénom, entreprise pour laquelle ils travaillent ou encore leur année de
                        mandat.
                        Si vous avez des remarques ou suggestions, merci d'en faire par à
                        <CopyContent className="contact" textToCopy="contact@etic-insa.com"/>
                    </div>
                </section>
            </div>
        </React.Fragment>
    );
};


export default Help;
