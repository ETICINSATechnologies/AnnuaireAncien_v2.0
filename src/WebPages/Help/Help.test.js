import React from 'react';

import Help from './Help';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Auth from "../../Components/Auth/Auth";
import {BrowserRouter as Router} from "react-router-dom";



Enzyme.configure({adapter: new Adapter()});

describe( 'Help component', () => {
    it('renders without crashing', () => {
            const tree = renderer.create(<Help />).toJSON();
            expect(tree).toMatchSnapshot();
        }
    );

    it('Help unit test', () => {
        expect(
            Enzyme.shallow(<Help/>)
                .find('div.Help')
                .length
        ).toBe(1)
    });

    it('Help button test', () => {
            const wrapper = Enzyme.shallow(<Help />);
            expect( wrapper.find('Nav').length ).toBe(1);
        }
    );

    it('Help not connected snapshot test', () => {
        Auth.isConnected = jest.fn(() => false);
        const component = renderer.create(<Help/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });

    it('Help connected snapshot test', () => {
        Auth.isConnected = jest.fn(() => true);
        const component = renderer.create(<Help/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});