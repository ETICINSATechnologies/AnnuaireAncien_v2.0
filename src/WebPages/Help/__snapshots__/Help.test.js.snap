// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help component Help connected snapshot test 1`] = `
Array [
  <header
    className="Header"
  >
    <a
      className="header_a"
      href="/"
    >
      <img
        alt=""
        className="home_logo"
        src="home_logo.png"
      />
      <h1
        className="title"
      >
         Annuaire des Anciens
      </h1>
    </a>
  </header>,
  <nav
    className="Nav"
  >
    <div
      className="Menu"
    >
      <a
        className="button"
        href="/"
      >
        accueil
      </a>
      <a
        className="button"
        href="/search"
      >
        rechercher
      </a>
      <a
        className="button"
        href="/profile"
      >
        mon profil
      </a>
      <a
        className="button"
        href="/"
        onClick={[Function]}
      >
        se déconnecter
      </a>
    </div>
    <a
      className="help_link"
      href="/help"
    >
      <img
        alt=""
        className="help_icon"
        src="help_icon.png"
      />
    </a>
  </nav>,
  <div
    className="Help"
  >
    <section
      className="help_section"
    >
      <div
        className="div1"
      >
        Cette application a été développée par
      </div>
      <a
        className="div2"
        href="http://etic-insa.com/"
      >
         ETIC INSA Technologies 
      </a>
      <div
        className="div3"
      >
        Elle permet de rechercher les anciens membres de la Junior en fonction de divers critères comme leur nom, prénom, entreprise pour laquelle ils travaillent ou encore leur année de mandat Si vous avez des remarques ou suggestions, merci d'en faire par à
        <button
          className="div4"
        >
           contact@etic-insa.com 
        </button>
        <div
          className="div5"
        >
           adresse copié !
        </div>
      </div>
    </section>
  </div>,
]
`;

exports[`Help component Help not connected snapshot test 1`] = `
Array [
  <header
    className="Header"
  >
    <a
      className="header_a"
      href="/"
    >
      <img
        alt=""
        className="home_logo"
        src="home_logo.png"
      />
      <h1
        className="title"
      >
         Annuaire des Anciens
      </h1>
    </a>
  </header>,
  <nav
    className="Nav"
  >
    <div
      className="Menu"
    >
      <a
        className="button"
        href="/"
      >
        accueil
      </a>
      <a
        className="button"
        href="/connection"
      >
        se connecter
      </a>
    </div>
    <a
      className="help_link"
      href="/help"
    >
      <img
        alt=""
        className="help_icon"
        src="help_icon.png"
      />
    </a>
  </nav>,
  <div
    className="Help"
  >
    <section
      className="help_section"
    >
      <div
        className="div1"
      >
        Cette application a été développée par
      </div>
      <a
        className="div2"
        href="http://etic-insa.com/"
      >
         ETIC INSA Technologies 
      </a>
      <div
        className="div3"
      >
        Elle permet de rechercher les anciens membres de la Junior en fonction de divers critères comme leur nom, prénom, entreprise pour laquelle ils travaillent ou encore leur année de mandat Si vous avez des remarques ou suggestions, merci d'en faire par à
        <button
          className="div4"
        >
           contact@etic-insa.com 
        </button>
        <div
          className="div5"
        >
           adresse copié !
        </div>
      </div>
    </section>
  </div>,
]
`;

exports[`Help component renders without crashing 1`] = `
Array [
  <header
    className="Header"
  >
    <a
      className="header_a"
      href="/"
    >
      <img
        alt=""
        className="home_logo"
        src="home_logo.png"
      />
      <h1
        className="title"
      >
         Annuaire des Anciens
      </h1>
    </a>
  </header>,
  <nav
    className="Nav"
  >
    <div
      className="Menu"
    >
      <a
        className="button"
        href="/"
      >
        accueil
      </a>
      <a
        className="button"
        href="/connection"
      >
        se connecter
      </a>
    </div>
    <a
      className="help_link"
      href="/help"
    >
      <img
        alt=""
        className="help_icon"
        src="help_icon.png"
      />
    </a>
  </nav>,
  <div
    className="Help"
  >
    <section
      className="help_section"
    >
      <div
        className="div1"
      >
        Cette application a été développée par
      </div>
      <a
        className="div2"
        href="http://etic-insa.com/"
      >
         ETIC INSA Technologies 
      </a>
      <div
        className="div3"
      >
        Elle permet de rechercher les anciens membres de la Junior en fonction de divers critères comme leur nom, prénom, entreprise pour laquelle ils travaillent ou encore leur année de mandat Si vous avez des remarques ou suggestions, merci d'en faire par à
        <button
          className="div4"
        >
           contact@etic-insa.com 
        </button>
        <div
          className="div5"
        >
           adresse copié !
        </div>
      </div>
    </section>
  </div>,
]
`;
